### PR TITLE
(feature) O3-2966: Appointments: Support custom form when checking in…

### PR DESCRIPTION
**Pull request by mogoodrich**
… and checking out

See; https://openmrs.atlassian.net/browse/O3-2966

## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

### CODE REVIEW

**Comment by mogoodrich** [comment]
Continuing the "nesting" pull requests (this is the last one, I promise!) this one builds off O3-2965

**Review by mseaton — APPROVED** [review]
LGTM

File: packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx [hunk 1]
```diff
@@ -34,13 +35,17 @@ const AppointmentActions: React.FC<AppointmentActionsProps> = ({ appointment })
...
   const handleCheckout = () => {
-    const dispose = showModal('end-visit-dialog', {
-      closeModal: () => {
-        mutateVisit();
-        dispose();
-      },
-      patientUuid,
-    });
+    if (checkOutButton.customUrl) {
+      navigate({ to: checkOutButton.customUrl, templateParams: { patientUuid, appointmentUuid: appointment.uuid } });
```
Comment (chibongho) [inline packages/esm-appointments-app/src/appointments/common-components/appointments-actions.component.tsx:39]: Be explicit in the config description (in config-schema.ts) that the the customUrl supports those 2 template params.

**Review by chibongho — APPROVED** [review]

**Comment by ibacher** [comment]
Is it intended to merge this into the other PR branch?

**Comment by mogoodrich** [comment]
Yeah, @ibacher  I've basically got three different PRs that build on each other, going to merge O3-2966 into O3-2965 and then both those into O3-2964 and then in master.  They all build off each other but made it easier to break the commits into smaller chunks for review.

**Comment by mogoodrich** [comment]
Thanks @mseaton  and @chibongho , merging!  @donaldkibet  and @ibacher  post-commit reviews still welcome

**Pull request merged**